### PR TITLE
Bookings functionality

### PR DESCRIPTION
**Pull request by khan2597**
Contributors : @Slongboro and @marcb1988

### CODE REVIEW

File: lib/booking.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require 'pg'
+
+class Booking
```
Comment (marcb1988) [inline lib/booking.rb:3]: Yup, this is a class alright 👍

File: lib/booking.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require 'pg'
+
+class Booking 
+    def self.create(listing_id, guest_username, date_from, date_to)
```
Comment (marcb1988) [inline lib/booking.rb:4]: Good use of methods 👍

File: lib/booking.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        connection = Database.connect
+        connection.exec_params("INSERT INTO booking (listing_id, guest_username, date_from, date_to) VALUES ($1, $2, $3, $4)",
```
Comment (marcb1988) [inline lib/booking.rb:6]: good use of SQL parameters

File: lib/booking.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+                               [listing_id, guest_username, date_from, date_to])
+    end
+
+    def self.get_booking_by_guest_username(username)
+        connection = Database.connect
+        booking = connection.exec_params("SELECT * FROM booking WHERE guest_username=$1", [username]).first
```
Comment (marcb1988) [inline lib/booking.rb:12]: good use of filtering

File: spec/booking_spec.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+require 'booking'
+require 'property'
+require 'user'
+require 'pg'
+
+describe Booking do
+    it 'creates an instance of the booking class' do
```
Comment (marcb1988) [inline spec/booking_spec.rb:7]: which is why sausage has the same behaviour as weather

File: spec/setup_test_database.rb [hunk 6]
```diff
@@ -5,5 +5,5 @@ def setup_test_database
 
   connection = PG.connect(dbname: "makersbnb_test")
 
-  connection.exec("TRUNCATE users, listings;")
+  connection.exec("TRUNCATE users, listings, booking;")
```
Comment (marcb1988) [inline spec/setup_test_database.rb:8]: well done for catching that!

**Review by marcb1988 — APPROVED** [review]

**Review by thorpelawrence — APPROVED** [review]

**Pull request merged**